File: new/src/Pan3d/skill/key/SkillKey.ts
module Pan3d {
    export class SkillKey {
        public time: number = 0;
        public particle: CombineParticle;
        public removeCallFun: Function;
        public scene3D:Scene3D;
        constructor(value:Scene3D ) {
            this.scene3D=value;
        }

        public addToRender(): void {
            if (!this.particle) {
                return;
            }
            this.particle.reset();
            this.particle.sceneVisible = true

           this.scene3D. particleManager .addParticle(this.particle);
        }



        public setInfo(obj: SkillKeyVo): void {
            this.time = obj.frame * Scene3D.frameTime;

            this.particle =  this.scene3D. particleManager .getParticleByte(this.scene3D.fileRoot + obj.url);


        }

        public reset(): void {
            //this.time = 0;
            this.particle.reset();
            this.scene3D. particleManager .removeParticle(this.particle);
        }

        public destory(): void {
            // this.particle.destory();
            this.particle = null;
            this.removeCallFun = null;
        }

    }


}

File: android/android/app/src/main/java/com/pan3d/skill/key/SkillKey.java
package com.pan3d.skill.key;

import com.pan3d.base.CallBack;
import com.pan3d.base.Scene_data;
import com.pan3d.display.particle.CombineParticle;
import com.pan3d.filemodel.ParticleManager;
import com.pan3d.scene.Scene3D;
import com.pan3d.skill.vo.SkillKeyVo;

public class SkillKey {
    public float time ;
    public CombineParticle particle;
    public CallBack removeCallFun;

    public Scene3D scene3D;
    public SkillKey(Scene3D val){
        scene3D=val;
    }
    public void addToRender(ParticleManager val) {
        if (this.particle==null) {
            return;
        }

        this.particle.reset();
        this.particle.sceneVisible = true;
        scene3D.particleManager.addParticle(this.particle);
    }

    public void setInfo(SkillKeyVo obj) {
        this.time = obj.frame * Scene_data.frameTime;
        this.particle =scene3D.particleManager.getParticleByte( obj.url);


    }
}
